src/pointcloud_to_laserscan_node.cpp

```diff
@@ -225,9 +243,80 @@ void PointCloudToLaserScanNode::cloudCallback(
     int index = (angle - scan_msg->angle_min) / scan_msg->angle_increment;
     if (range < scan_msg->ranges[index]) {
       scan_msg->ranges[index] = range;
+      v_pointcloud_index.at(index) = pointcloud_index;
     }
   }
-  pub_->publish(std::move(scan_msg));
+
+  laserscan_pub_->publish(std::move(scan_msg));
+
+  pcl::PointCloud<pcl::PointXYZ> pcl_pointcloud;
+  sensor_msgs::PointCloud2ConstIterator<float> iter_x(*cloud_msg, "x"), iter_y(*cloud_msg, "y"),
+    iter_z(*cloud_msg, "z");
+  for (size_t i = 0; i < v_pointcloud_index.size(); ++i) {
```

Comment:
Can we replace this with `range-based-for`?

Reply:
Fixed at https://github.com/tier4/pointcloud_to_laserscan/pull/1/commits/2035b1be800f727c607d91cc87d82345cd631398